### PR TITLE
Vampires can now gain blood for a minute after someone succumbs to death

### PR DESCRIPTION
**Pull request by Qwertytoforty**
## What Does This PR Do

Adds a trait to people who succumb to death for one minute.
People with this trait can be drained by vampires like they are living.

## Why It's Good For The Game

We already tell people off for suiciding when caught by antagonists. We don't need people being slightly beat up by a vampire and succumbing to deny vampires blood. If a vampire fails to heal them out of crit, and  they die, that's on the vampire. If someone instantly enters crit and succumbs, thats not the vampires fault, and not something that should be happening

## Testing

Sucked off a skrell after succumbing as them. Confirmed that after a minute I did not get usable blood.

## Changelog
:cl:
tweak: Vampires can now gain blood for a minute after someone succumbs to death
/:cl:

### CODE REVIEW

File: code/modules/antagonists/vampire/vamp_datum.dm [hunk 1]
```diff
@@ -135,7 +135,7 @@
 				continue
 
 
-		if(H.stat < DEAD)
+		if(H.stat < DEAD || HAS_TRAIT(H, TRAIT_RECENTLY_SUCCUMBED))
```
Comment (DGamerL) [inline code/modules/antagonists/vampire/vamp_datum.dm:138]:
```suggestion
		if(H.stat != DEAD || HAS_TRAIT(H, TRAIT_RECENTLY_SUCCUMBED))
```

File: code/modules/mob/living/living.dm [hunk 2]
```diff
@@ -320,6 +320,11 @@
 	else
 		death()
 	to_chat(src, "<span class='notice'>You have given up life and succumbed to death.</span>")
+	ADD_TRAIT(src, TRAIT_RECENTLY_SUCCUMBED, "succumb")
+	addtimer(CALLBACK(src, PROC_REF(remove_succumbed)), 1 MINUTES)
+
+/mob/living/proc/remove_succumbed()
+	REMOVE_TRAIT(src, TRAIT_RECENTLY_SUCCUMBED, "succumb")
```
Comment (DGamerL) [inline code/modules/mob/living/living.dm:328]:
```suggestion
```

File: code/_globalvars/traits.dm [hunk 3]
```diff
@@ -87,7 +87,8 @@ GLOBAL_LIST_INIT(traits_by_type, list(
 		"TRAIT_BADASS" = TRAIT_BADASS,
 		"TRAIT_FORCED_STANDING" = TRAIT_FORCED_STANDING,
 		"TRAIT_NOSLIP" = TRAIT_NOSLIP,
-		"TRAIT_MAGPULSE" = TRAIT_MAGPULSE
+		"TRAIT_MAGPULSE" = TRAIT_MAGPULSE,
+		"TRAIT_RECENTLY_SUCCUMBED " = TRAIT_RECENTLY_SUCCUMBED
```
Comment (lewcc) [inline code/_globalvars/traits.dm:91]:
```suggestion
		"TRAIT_RECENTLY_SUCCUMBED" = TRAIT_RECENTLY_SUCCUMBED
```

**Review by lewcc — COMMENTED** [review]
One note. However, I'd recommend making this a status effect (much like how revivability works w/ CPR rework), so it automatically expires after some time.

**Review by DGamerL — APPROVED** [review]

**Comment by Probablyjay19** [comment]
You did w h a t during testing

**Review by lewcc — APPROVED** [review]
Thanks for making it a status effect, looks good

**Pull request merged**